FHICORC.Backend/FHICORC.Application.Services/RevocationService.cs

```diff
@@ -53,6 +60,140 @@ public RevocationService(ILogger<RevocationService> logger, CoronapassContext co
             {
                 SuperBatches = superBatchList
             };           
-        }     
+        }
+
+        public void UploadHashes(List<string> hashList)
+        {
+            var revokedHashList = FetchHashes();
+            var newHashList = ReturnUniqueHashes(hashList, revokedHashList);
+            var batchItem = FetchSmallestBatchItem();
+
+            Guid newGuid = Guid.NewGuid();
+            bool expiresMatch = false;
+            bool firstIteration = true;
+            int counter = 1;
+            int batchItemCount = 0;
+            int countHashList = newHashList.Count;
+
+            if (batchItem != null)
+            {
+                expiresMatch = ExpiryDate == batchItem.Expires.Date;
+                batchItemCount = batchItem.Count;
+            }
+
+            foreach (var hash in newHashList.Select((value, index) => new { value, index }))
```

Comment:
I dont think we have any performance requirements for this, but if we do this for loop needs to change to not call the database for every hash & batch created.